src/main/java/com/gregtechceu/gtceu/common/cover/StorageCover.java

```diff
@@ -0,0 +1,51 @@
+package com.gregtechceu.gtceu.common.cover;
+
+import com.gregtechceu.gtceu.api.capability.ICoverable;
+import com.gregtechceu.gtceu.api.cover.CoverBehavior;
+import com.gregtechceu.gtceu.api.cover.CoverDefinition;
+import com.gregtechceu.gtceu.api.cover.IUICover;
+
+import com.lowdragmc.lowdraglib.gui.widget.SlotWidget;
+import com.lowdragmc.lowdraglib.gui.widget.Widget;
+import com.lowdragmc.lowdraglib.gui.widget.WidgetGroup;
+import com.lowdragmc.lowdraglib.misc.ItemStackTransfer;
+import com.lowdragmc.lowdraglib.syncdata.annotation.DescSynced;
+import com.lowdragmc.lowdraglib.syncdata.annotation.Persisted;
+
+import net.minecraft.core.Direction;
+
+import org.jetbrains.annotations.NotNull;
+
+public class StorageCover extends CoverBehavior implements IUICover {
+
+    @Persisted
+    @DescSynced
+    public final ItemStackTransfer inventory;
+    private int inventorysize = 18;
+    private int xoffset = 7;
+    private int yoffset = 7;
+
+    public StorageCover(@NotNull CoverDefinition definition, @NotNull ICoverable coverableView,
+                        @NotNull Direction attachedSide) {
+        super(definition, coverableView, attachedSide);
+        inventory = new ItemStackTransfer(inventorysize) {
+
+            @Override
+            public int getSlotLimit(int slot) {
+                return 1;
+            }
+        };
+    }
+
+    @Override
+    public Widget createUIWidget() {
+        final var group = new WidgetGroup(0, 0, 176, 85);
+        for (int slot = 0; slot < 18; slot++) {
```

Comment:
could check inventorySize here instead of hardcoding for 18 slots.

Reply:
also, this method has 2x as many empty lines as it needs.

Reply:
illl fix this when i add a tab which means burnout has to be over 👍